src/test/java/test/dependent/DependentTest.java

```diff
@@ -181,8 +181,7 @@ public void simpleCyclingDependencyShouldWork(
         assertThat(listener.getInvokedMethodNames()).containsExactly(runMethods);
         assertThat(listener.getSucceedMethodNames()).containsExactly(runMethods);
     } else {
-        // When running parallel, invoke order is consistent, but succeed order isn't.
```

Comment:
The test case is useless with this assertion.
We need to assert that the invoked method order is right.
If any orders are possible, maybe it is possible to describe an "or" logic.

Reply:
Reverted this change and i went back to tracing this change to this 
https://github.com/cbeust/testng/issues/1380#issuecomment-286880800

If I stumble into this issue again, then I will have another look at this.

Reply:
@juherr - Please ignore the previous comment. I encountered the problem once again. So refactored the code to split the parallel and sequential scenarios as two different tests and added. a predicate based matcher for evaluating the parallel combinations.